examples/getstarted/src/admin/preview/dummy-preview.jsx

```diff
@@ -1,14 +1,119 @@
 import * as React from 'react';
 import { useParams, useLoaderData, useRevalidator } from 'react-router-dom';
+import { BlocksRenderer } from '@strapi/blocks-react-renderer';
 
 import { Page, Layouts } from '@strapi/admin/strapi-admin';
-import { Grid, Flex, Typography, JSONInput, Box } from '@strapi/design-system';
+import { Grid, Flex, Typography, JSONInput, Box, IconButton } from '@strapi/design-system';
+import { ChevronDown, ChevronRight } from '@strapi/icons';
 
 const filterAttributes = (item) => {
   const excludedKeys = ['documentId', 'id', 'createdAt', 'updatedAt', 'publishedAt'];
   return Object.entries(item).filter(([key]) => !excludedKeys.includes(key));
 };
 
+const ToggleableContainer = ({ headerText, children, isEmpty = false }) => {
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  return (
+    <Flex direction="column" alignItems="flex-start" gap={1}>
+      <Flex
+        tag="button"
+        alignItems="center"
+        gap={1}
+        onClick={() => setIsExpanded(!isExpanded)}
+        paddingTop={1}
+        paddingBottom={1}
+        marginLeft={-1}
+        disabled={isEmpty}
+        color="neutral600"
+      >
+        {isExpanded ? <ChevronDown /> : <ChevronRight />}
+        <Typography variant="omega">{headerText}</Typography>
+      </Flex>
+      {isExpanded && (
+        <Box
+          borderWidth="0 0 0 2px"
+          borderColor="neutral200"
+          paddingLeft={3}
+          paddingTop={1}
+          paddingBottom={1}
+        >
+          {children}
+        </Box>
+      )}
+    </Flex>
+  );
+};
+
+const NestedValue = ({ value, level = 0, arrayIndex = undefined }) => {
+  const isBlocksField =
+    Array.isArray(value) &&
+    value.length > 0 &&
+    value.every((item) => 'type' in item && 'children' in item);
+
+  if (isBlocksField) {
+    return (
+      <Flex direction="column" alignItems="flex-start" fontSize="1.4rem" gap={2}>
+        <BlocksRenderer content={value} />
+      </Flex>
+    );
+  }
+
+  if (Array.isArray(value)) {
+    return (
+      <ToggleableContainer
+        headerText={`Array (${value.length} items)`}
+        isEmpty={value.length === 0}
+      >
+        <Flex direction="column" gap={2}>
+          {value.map((item, index) => (
+            <Box key={index}>
+              {typeof item === 'object' && item !== null ? (
+                <NestedValue value={item} level={level + 1} arrayIndex={index} />
+              ) : (
+                <Flex direction="column" gap={1}>
+                  <Typography variant="pi" textColor="neutral500">
+                    [{index}]
+                  </Typography>
+                  <NestedValue value={item} level={level + 1} arrayIndex={index} />
+                </Flex>
+              )}
+            </Box>
+          ))}
+        </Flex>
+      </ToggleableContainer>
+    );
+  }
+
+  if (typeof value === 'object' && value !== null) {
```

Comment:
idk if it's valuable for you but I see you doing this same check in a lot of places, maybe time for an util?